includes/DataObjects/Request.php

```diff
@@ -263,12 +271,12 @@ public function hasComments()
 			return true;
 		}
 
-		$commentsQuery = $this->dbObject->prepare("SELECT COUNT(*) as num FROM comment where request = :id;");
-		$commentsQuery->bindValue(":id", $this->id);
+		if (is_null(self::$commentsQuery)) self::$commentsQuery = $this->dbObject->prepare("SELECT COUNT(*) as num FROM comment where request = :id;");
+		self::$commentsQuery->bindValue(":id", $this->id);
 
-		$commentsQuery->execute();
+		self::$commentsQuery->execute();
 
-		$this->hasComments = ($commentsQuery->fetchColumn() != 0);
+		self::$commentsResults[$this->id] = $this->hasComments = (self::$commentsQuery->fetchColumn() != 0);
```

Comment:
It's much easier to read and hence understand and debug if you only put one assignment per line. Thus:

```php
$this->hasComments = (self::$commentsQuery->fetchColumn() != 0);
self::$commentsResults[$this->id] = $this->hasComments;
```